ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/KsqlResource.java

```diff
@@ -187,36 +123,64 @@ public KsqlResource(
       final Duration distributedCmdResponseTimeout,
       final ActivenessRegistrar activenessRegistrar
   ) {
-    this.ksqlConfig = Objects.requireNonNull(ksqlConfig, "ksqlConfig");
+    this(
+        ksqlConfig,
+        ksqlEngine,
+        serviceContext,
+        commandQueue,
+        distributedCmdResponseTimeout,
+        activenessRegistrar,
+        new RequestValidator(
+            CUSTOM_VALIDATORS,
+            OPTIONAL_VALIDATION,
+            () -> new EngineSnapshot(ksqlEngine.createSandbox()),
+            serviceContext.getSchemaRegistryClient(),
+            ksqlConfig),
+        new RequestHandler(
+            CUSTOM_EXECUTORS,
+            KsqlResource::shouldSynchronize,
+            new DistributingExecutor(commandQueue, distributedCmdResponseTimeout),
+            ksqlEngine,
+            ksqlConfig,
+            serviceContext,
+            commandQueue,
+            distributedCmdResponseTimeout)
+    );
+  }
+
+  KsqlResource(
+      final KsqlConfig ksqlConfig,
+      final KsqlEngine ksqlEngine,
+      final ServiceContext serviceContext,
+      final CommandQueue commandQueue,
+      final Duration distributedCmdResponseTimeout,
+      final ActivenessRegistrar activenessRegistrar,
+      final RequestValidator requestValidator,
+      final RequestHandler requestHandler
+  ) {
+    Objects.requireNonNull(ksqlConfig, "ksqlConfig");
+    Objects.requireNonNull(serviceContext, "serviceContext");
     this.ksqlEngine = Objects.requireNonNull(ksqlEngine, "ksqlEngine");
-    this.serviceContext = Objects.requireNonNull(serviceContext, "serviceContext");
     this.commandQueue = Objects.requireNonNull(commandQueue, "commandQueue");
     this.distributedCmdResponseTimeout =
         Objects.requireNonNull(distributedCmdResponseTimeout, "distributedCmdResponseTimeout");
     this.activenessRegistrar =
         Objects.requireNonNull(activenessRegistrar, "activenessRegistrar cannot be null.");
+    this.validator = requestValidator;
+    this.handler = requestHandler;
   }
 
   @POST
   @Path("/terminate")
   public Response terminateCluster(final ClusterTerminateRequest request) {
-    final KsqlEntityList result = new KsqlEntityList();
-
     ensureValidPatterns(request.getDeleteTopicList());
-
     try {
-      result.add(distributeStatement(
-          PreparedStatement.of(TerminateCluster.TERMINATE_CLUSTER_STATEMENT_TEXT,
-              new TerminateCluster()),
-          request.getStreamsProperties()
-      ));
-
-      return Response.ok(result).build();
+      return Response.ok(
```

Comment:
Shame we can't move terminate onto its own resource handler.

Reply:
😢